.github/workflows/test.yml

```diff
@@ -4,7 +4,7 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.18.x]
+        go-version: [1.21.x]
```

Comment:
Just confirming, we are setting go.mod to `1.20`, but here we are testing with the newest stable intentionally.

Given it's a library and not a cli, I don't think they are required to match. Just wanted to confirm intent and not a typo.

Reply:
Correct, this is intentional. Updating `go.mod` to >= 1.21.0 means that the library requires that version as a minimum. But, that seems a bit overreaching. So, we keep it back to 1.20 so we aren't forcing users of this library to upgrade.

Reply:
(See: https://tip.golang.org/doc/go1.21#tools for example)